#  p-value 

<img src='http://infoso.kr/wp/wp-content/uploads/2016/03/P_Value-1020x841.jpeg'>
##  p-value란?

**p-value (유의확률)**는 통계적 가설 검정에서  
**귀무가설이 참일 때, 지금 관측된 결과와 같거나 더 극단적인 결과가 나올 확률**입니다.

---


###  예: 동전이 공정한지 실험

- **귀무가설 (H₀)**: 이 동전은 공정하다 (앞면 나올 확률 = 0.5)
- **실험 결과**: 동전을 10번 던졌더니 앞면이 **9번** 나왔다!

 우리는 이렇게 묻습니다:

> “공정한 동전이라면, 10번 중 9번 이상 앞면이 나올 확률은 얼마나 될까?”

이 확률이 바로 **p-value**입니다.

---

##  p-value 해석 방법

| p-value 값        | 해석 설명                                                  |
|-------------------|------------------------------------------------------------|
| **작다 (예: 0.01)** | 귀무가설 하에서는 이런 결과가 거의 안 일어남 → 귀무가설 **기각** |
| **크다 (예: 0.6)**  | 귀무가설 하에서도 이런 결과는 자주 일어남 → 귀무가설 **유지** |

---

##  일반적인 기준

- **유의수준 (α)**: 보통 0.05 사용
- **p-value < 0.05** → 귀무가설 기각 (차이가 통계적으로 유의함)
- **p-value ≥ 0.05** → 귀무가설 기각하지 않음 (차이가 통계적으로 유의하지 않음)

---

##  요약

- **p-value는 "우연히 이런 결과가 나올 확률"**입니다.
- **작을수록** → "이건 우연이 아니야!"라고 주장할 수 있습니다.
- 단, p-value는 **귀무가설이 맞다고 가정**한 상태에서 해석해야 합니다.
- **진실을 말해주는 게 아니라**, **귀무가설 하에서 결과가 얼마나 이상한지를 측정**할 뿐입니다.

---



##  p-value (유의 확률) 정리

p-value(유의 확률)는 통계에서 가설 검정을 할 때, **귀무가설(null hypothesis)이 참일 때 관찰된 데이터가 발생할 확률**을 의미합니다. 쉽게 말해, **"이러한 데이터가 우연히 나올 가능성"**을 나타냅니다.

* **귀무가설 ($\text{H}_0$):** 보통 "**차이가 없다**" 또는 "**효과가 없다**"는 가설입니다.
* **대립가설 ($\text{H}_1$):** "**차이가 있다**" 또는 "**효과가 있다**"는 가설입니다.

---

### p-value 해석

p-value는 다음과 같이 해석됩니다.

* **p-value가 작을수록** (일반적으로 $0.05$ 미만), 귀무가설이 사실일 가능성이 낮다는 뜻이므로, 이를 **기각**하고 대립가설을 **채택**할 수 있습니다.
* **p-value가 크면** 귀무가설을 기각할 근거가 부족하므로, 귀무가설을 **유지**합니다.

---

### 예시

예를 들어, **p-value가 $0.03$**이라면, 귀무가설이 참일 때 이러한 데이터를 얻을 확률이 $3\%$라는 뜻입니다. 이 경우, 보통은 귀무가설을 기각하고 대립가설을 채택합니다.

---


##  단측 검정 vs. 양측 검정

단측 검정(One-Tailed Test)과 양측 검정(Two-Tailed Test)의 차이는 \*\*"우리가 알고 싶은 차이의 방향"\*\*이 정해져 있는지 여부에 달려있습니다.

쉽게 말해, 우리가 **특정 방향(더 크다, 더 작다)만을 예상**하고 검증할 때는 **단측 검정**을 사용하고, **방향에 상관없이 단순히 차이가 있는지**만 알고 싶을 때는 **양측 검정**을 사용합니다.

-----

### 1\. 양측 검정 (Two-Tailed Test)

양측 검정은 "차이가 있다"는 것을 검증하며, 그 차이가 **기준값보다 클 수도 있고 작을 수도 있는** 양쪽의 가능성을 모두 열어둡니다.

  * **대립가설 ($\text{H}_1$):** "A와 B는 **같지 않다**($\neq$)."
      * **예시:** "새로운 약의 효과는 기존 약의 효과와 **다르다**." (더 좋을 수도, 더 나쁠 수도 있음)
  * **기각역:**

<img src='https://encrypted-tbn3.gstatic.com/licensed-image?q=tbn:ANd9GcRc9zbnXfyazHc67I1QcW-S_y3Njexfh7QIiOLHkJRk4E2E8IaJmakfBROHcLHIUJ5gLy0_JKJ6tuNu2FGImwbZBeLaDiOSvsfQxYfP7AciZwIW48E'>


* 분포의 **양쪽 끝 꼬리**에 유의수준($\alpha$)을 반으로 나누어 기각 영역(Rejection Region)을 설정합니다. (예: $\alpha = 0.05$일 경우, 왼쪽 꼬리 $0.025$, 오른쪽 꼬리 $0.025$)
* 검정 통계량이 이 양쪽 기각역 중 **어느 한쪽**에 들어가면 귀무가설($\text{H}_0$: 차이가 없다)을 기각하고 대립가설을 채택합니다.


-----

### 2\. 단측 검정 (One-Tailed Test)

단측 검정은 **특정한 방향**의 차이만 검증하며, 관심 있는 방향(더 크다 또는 더 작다)만 살펴봅니다.

  * **대립가설 ($\text{H}_1$):** "A는 B보다 **크다**($>$)" **또는** "A는 B보다 **작다**($<$)."
      * **예시:** "새로운 약은 기존 약보다 **효과가 더 좋을 것이다**." (효과가 더 나쁜 경우는 관심 밖)
  * **기각역:**
      * 분포의 **한쪽 꼬리**(예상한 방향)에 유의수준($\alpha$) 전체를 기각 영역으로 설정합니다. (예: $\alpha = 0.05$ 전체를 한쪽 꼬리에 설정)
      * 검정 통계량이 이 **한쪽 기각역**에 들어가면 귀무가설을 기각하고 대립가설을 채택합니다.

-----

###  핵심 차이 요약

| 구분 | 양측 검정 (Two-Tailed) | 단측 검정 (One-Tailed) |
| :--- | :--- | :--- |
| **질문** | "차이가 **있는가**?" (방향 무관) | "더 **크거나** 더 **작은가**?" (방향 중요) |
| **대립가설 ($\text{H}_1$**) | $\neq$ (같지 않다) | $>$ (크다) 또는 $<$ (작다) |
| **기각역 위치** | 분포의 **양쪽 꼬리** ($\alpha/2$씩 분할) | 분포의 **한쪽 꼬리** ($\alpha$ 전체) |
| **검정력** | 단측보다 **낮음** (유의성 도달이 어려움) | 양측보다 **높음** (유의성 도달이 쉬움) |

**주의:** 데이터 분석 전에 연구 가설의 성격에 따라 미리 단측 또는 양측 검정을 결정해야 합니다. 데이터를 본 후 p-value를 유리하게 만들기 위해 선택하는 것은 통계적으로 부적절합니다.

-----

### 예시: p-value $0.05$에 해당하는 **임계 Z값 (Critical Z-value)**은 단측 검정과 양측 검정에서 다음과 같습니다. 이 Z값은 **표준정규분포**를 따른다고 가정했을 때의 값입니다.

---

### 1. 양측 검정 (Two-Tailed Test)

양측 검정에서 유의수준 $\alpha = 0.05$는 분포의 양쪽 꼬리(Two Tails)에 각각 $\alpha/2 = 0.025$씩 분할됩니다.

따라서, 누적 확률이 $0.025$와 $0.975$가 되는 Z값을 찾습니다.

* **Z값:** $\mathbf{\pm 1.96}$ (더 정확하게는 $\pm 1.95996...$)

> 귀무가설($\text{H}_0$)을 기각하기 위해서는, 검정 통계량 Z값이 $-1.96$보다 작거나 **또는** $+1.96$보다 커야 합니다.
> 

[Image of two-tailed test rejection region on a normal distribution]


---

### 2. 단측 검정 (One-Tailed Test)

단측 검정에서 유의수준 $\alpha = 0.05$는 분포의 한쪽 꼬리(One Tail)에 집중됩니다.

* **오른쪽 꼬리 검정 (Right-Tailed Test):**
    * 누적 확률이 $1 - 0.05 = 0.95$가 되는 Z값을 찾습니다.
    * **Z값:** $\mathbf{+1.645}$ (더 정확하게는 $1.64485...$)
* **왼쪽 꼬리 검정 (Left-Tailed Test):**
    * 누적 확률이 $0.05$가 되는 Z값을 찾습니다.
    * **Z값:** $\mathbf{-1.645}$ (더 정확하게는 $-1.64485...$)

> 귀무가설($\text{H}_0$)을 기각하기 위해서는, 검정 방향에 따라 Z값이 $\mathbf{+1.645}$보다 크거나 $\mathbf{-1.645}$보다 작아야 합니다.

---

###  요약 및 핵심 원리

| 구분 | 유의수준 ($\alpha$) 분할 | 임계 Z값 (Critical Z-value) |
| :--- | :--- | :--- |
| **양측 검정** | 양쪽 꼬리에 $\mathbf{0.025}$씩 분할 | $\mathbf{\pm 1.96}$ |
| **단측 검정** | 한쪽 꼬리에 $\mathbf{0.05}$ 전체 집중 | $\mathbf{+1.645}$ 또는 $\mathbf{-1.645}$ |

**핵심 원리:**
양측 검정은 기각역을 양쪽에 $\alpha/2$씩 나누어 놓기 때문에, 단측 검정 ($\alpha$ 전체)보다 Z값의 임계치가 **절댓값 기준으로 더 커집니다**. 즉, 양측 검정은 단측 검정보다 귀무가설을 기각하기가 **더 어렵습니다** (더 보수적입니다).

이러한 Z값을 사용하면, 계산된 **검정 통계량 Z값**이 이 임계 Z값을 벗어나는지 여부를 통해 가설을 기각할지 결정할 수 있습니다.


In [8]:
import numpy as np
from scipy.stats import norm

# 유의수준 (Alpha level)
alpha = 0.05

# 1. 양측 검정 임계 Z값 (Z-critical for Two-Tailed Test)
# 양쪽 꼬리 0.025씩 (누적 확률 0.975)
z_critical_two = norm.ppf(1 - alpha/2) # 1.95996... (약 1.96)

# 2. 단측 검정 임계 Z값 (Z-critical for One-Tailed Test)
# 한쪽 꼬리 0.05 (누적 확률 0.95)
z_critical_one = norm.ppf(1 - alpha) # 1.64485... (약 1.645)

print(f"양측 검정 임계 Z값: ±{z_critical_two:.3f}")
print(f"단측 검정 임계 Z값: ±{z_critical_one:.3f}")

# --- 가상의 검정 통계량 (Z-score) 예시 ---
# 이 값은 실제 데이터를 바탕으로 계산되었다고 가정합니다.
z_score_example = 2.05 
print(f"\n계산된 검정 통계량 Z: {z_score_example:.3f}")

양측 검정 임계 Z값: ±1.960
단측 검정 임계 Z값: ±1.645

계산된 검정 통계량 Z: 2.050


In [9]:
## 1. 양측 검정 (Two-Tailed Test)
print("\n## 1. 양측 검정 결과 (Z-critical: ±1.96)")

# 귀무가설 기각 조건: |Z| > Z_critical_two
if abs(z_score_example) > z_critical_two:
    # 2.05 > 1.96 이므로 기각
    print(f"Z={z_score_example:.3f}은(는) 임계 Z값 ±{z_critical_two:.3f}의 바깥에 있습니다.")
    print(" 귀무가설(차이가 없다)을 **기각**하고, 대립가설(차이가 있다)을 채택합니다.")
else:
    print(f"Z={z_score_example:.3f}은(는) 임계 Z값 ±{z_critical_two:.3f}의 안에 있습니다.")
    print(" 귀무가설을 **기각할 수 없습니다**.")


## 1. 양측 검정 결과 (Z-critical: ±1.96)
Z=2.050은(는) 임계 Z값 ±1.960의 바깥에 있습니다.
 귀무가설(차이가 없다)을 **기각**하고, 대립가설(차이가 있다)을 채택합니다.


In [10]:
## 2. 단측 검정 (Right-Tailed Test)
print("\n## 2. 단측 검정 결과 (Z-critical: +1.645)")

# 귀무가설 기각 조건: Z > Z_critical_one
if z_score_example > z_critical_one:
    # 2.05 > 1.645 이므로 기각
    print(f"Z={z_score_example:.3f}은(는) 임계 Z값 +{z_critical_one:.3f}보다 큽니다.")
    print(" 귀무가설(더 높지 않다)을 **기각**하고, 대립가설(더 높다)을 채택합니다.")
else:
    print(f"Z={z_score_example:.3f}은(는) 임계 Z값 +{z_critical_one:.3f}보다 작습니다.")
    print(" 귀무가설을 **기각할 수 없습니다**.")


## 2. 단측 검정 결과 (Z-critical: +1.645)
Z=2.050은(는) 임계 Z값 +1.645보다 큽니다.
 귀무가설(더 높지 않다)을 **기각**하고, 대립가설(더 높다)을 채택합니다.


- 동일한 검정 통계량 Z=2.05를 사용했음에도,양측 검정에서는 $1.96$을 넘기 때문에 유의미한 차이가 있다고 판단.
- 단측 검정에서는 $1.645$만 넘으면 되기 때문에 당연히 유의미한 차이가 있다고 판단.
- 이 예시를 통해 단측 검정이 양측 검정보다 귀무가설을 더 쉽게 기각할 수 있다는 것을 알 수 있다.

##  Z-검정 샘플

실제 데이터와 비슷한 시나리오를 가정하여, **어떤 제품의 평균 무게**가 기준치와 차이가 있는지 검정하는 예시.

###  시나리오

어떤 공장에서 생산되는 과자의 **표준 무게는 100g**이며, **모집단의 표준편차($\sigma$)는 5g**으로 알려져 있습니다. 오늘 생산된 과자 \*\*50개($n=50$)\*\*를 무작위로 추출하여 측정한 결과, **표본 평균($\bar{x}$)은 101.5g**이었습니다.

과연 이 **101.5g**이라는 결과가 **'우연히'** 나온 것인지, 아니면 **'실제로'** 과자 무게가 100g과 달라졌다고 볼 수 있는지 \*\*유의수준 $\alpha = 0.05$\*\*로 검정해 봅시다.

-----

###  코드 실행 및 Z-값 계산


In [ ]:

import numpy as np
from scipy.stats import norm

# 1. 시나리오 데이터
mu_0 = 100      # 귀무가설의 모평균 (기준 무게)
sigma = 5       # 모집단 표준편차 (알고 있다고 가정)
n = 50          # 표본 크기
x_bar = 101.5   # 표본 평균
alpha = 0.05    # 유의수준

# 2. 검정 통계량 Z 계산 (Z-score calculation)
# Z = (표본 평균 - 모평균) / (표준 오차)
# 표준 오차 (Standard Error) = sigma / sqrt(n)
std_error = sigma / np.sqrt(n)
z_score = (x_bar - mu_0) / std_error

print(f"표준 오차 (Standard Error): {std_error:.3f} g")
print(f"계산된 검정 통계량 Z-score: {z_score:.3f}")

표준 오차 (Standard Error): 0.707 g
계산된 검정 통계량 Z-score: 2.121


### 1\. 양측 검정 (Two-Tailed Test)

**가설:** 과자 무게가 **기준치(100g)와 다르다** ($\mu \neq 100g$).

In [14]:
# 양측 검정 임계 Z값 (Z-critical for Two-Tailed Test)
z_critical_two = norm.ppf(1 - alpha/2) # 1.96

print("\n## 1. 양측 검정 결과 (유의수준 0.05)")
print(f"임계 Z값 (±Z_critical): ±{z_critical_two:.3f}")
print(f"검정 Z값: {z_score:.3f}")

if abs(z_score) > z_critical_two:
    print(f" {z_score:.3f} > {z_critical_two:.3f} 이므로, 귀무가설 **기각**")
    print("해석: 과자 무게는 통계적으로 유의미하게 100g과 **다르다**고 결론 내릴 수 있습니다.")
else:
    print(" 귀무가설 기각 **실패**")


## 1. 양측 검정 결과 (유의수준 0.05)
임계 Z값 (±Z_critical): ±1.960
검정 Z값: 2.121
 2.121 > 1.960 이므로, 귀무가설 **기각**
해석: 과자 무게는 통계적으로 유의미하게 100g과 **다르다**고 결론 내릴 수 있습니다.


**결과 해석:**
계산된 Z값 $2.121$은 양측 임계값 $\pm 1.96$보다 크기 때문에, 귀무가설($\text{H}_0$: 무게는 100g이다)을 **기각**합니다. 즉, 새로운 과자 배치의 평균 무게는 **100g과 유의미하게 다릅니다.**


### 2\. 단측 검정 (One-Tailed Test)

**가설:** 과자 무게가 **기준치(100g)보다 무겁다** ($\mu > 100g$).

In [15]:
# 단측 검정 임계 Z값 (Z-critical for Right-Tailed Test)
z_critical_one = norm.ppf(1 - alpha) # 1.645

print("\n## 2. 단측 검정 결과 (유의수준 0.05)")
print(f"임계 Z값 (+Z_critical): +{z_critical_one:.3f}")
print(f"검정 Z값: {z_score:.3f}")

if z_score > z_critical_one:
    print(f" {z_score:.3f} > {z_critical_one:.3f} 이므로, 귀무가설 **기각**")
    print("해석: 과자 무게는 통계적으로 유의미하게 100g보다 **무겁다**고 결론 내릴 수 있습니다.")
else:
    print(" 귀무가설 기각 **실패**")


## 2. 단측 검정 결과 (유의수준 0.05)
임계 Z값 (+Z_critical): +1.645
검정 Z값: 2.121
 2.121 > 1.645 이므로, 귀무가설 **기각**
해석: 과자 무게는 통계적으로 유의미하게 100g보다 **무겁다**고 결론 내릴 수 있습니다.



**결과 해석:**
계산된 Z값 $2.121$은 단측 임계값 $1.645$보다 훨씬 크기 때문에, 귀무가설($\text{H}_0$: 무게가 100g보다 무겁지 않다)을 **기각**합니다. 즉, 새로운 과자 배치의 평균 무게는 **100g보다 무겁습니다.**

## 표준 오차(Standard Error)의 공식.

$$\text{표준 오차} = \frac{\sigma}{\sqrt{n}}$$

여기서 $\sigma$는 **모집단의 표준편차**이고, $n$은 **표본의 크기**입니다.


---


### 1. 표준 오차란

표준 오차(Standard Error, SE)는 우리가 **모집단**에서 추출한 **표본 평균**이 모집단의 **진짜 평균(모평균)**으로부터 **평균적으로 얼마나 떨어져 있는지를** 나타내는 척도입니다.

* 우리가 표본을 **한 번만** 뽑는 것이 아니라, 같은 크기의 표본을 **여러 번** 뽑아서 평균을 낼 수 있습니다.
* 이때, 표본 평균들은 서로 조금씩 다를 것입니다. 이 **'표본 평균들의 분포'**가 바로 **표본 분포**입니다.
* 표준 오차는 이 **표본 평균들의 분포가 얼마나 퍼져 있는지를 나타내는 표준편차**입니다.

### 2. 표준 오차가 표본 크기($n$)에 반비례하는 이유 ($\sqrt{n}$)

표준 오차 공식에서 표본 크기 $n$이 **분모**에 있고, 심지어 **제곱근($\sqrt{n}$)을 씌운 값**으로 나누어 주는 이유는 다음과 같습니다.

* **표본 크기가 커지면 정확도가 높아집니다:** 상식적으로 표본을 많이 뽑을수록(예: 10명 대신 1000명에게 설문) 모집단의 진짜 평균에 **더 가까운** 결과가 나올 것입니다.
* **표본 평균들의 오차가 줄어듭니다:** 즉, 표본 크기 $n$이 커질수록 우리가 뽑은 표본 평균($\bar{x}$)들이 진짜 모평균($\mu$) 주변에 **더 촘촘하게** 모이게 됩니다.
* 분포가 촘촘해진다는 것은 **분산(퍼짐 정도)**이 줄어든다는 의미이며, 이를 나타내는 **표준 오차도 줄어듭니다**.

**결론:** 표본 크기($n$)가 커지면 **분모**가 커지므로, 표준 오차는 **작아집니다**.

### 3. 표준 오차가 모집단 표준편차($\sigma$)에 비례하는 이유 ($\sigma$)

공식에서 모집단 표준편차 $\sigma$가 **분자**에 있는 이유는 다음과 같습니다.

* **모집단이 원래 들쑥날쑥하면:** 모집단의 데이터 자체가 넓게 퍼져 있다면 ($\sigma$가 크다면), 우리가 어떤 표본을 뽑더라도 그 표본의 평균이 진짜 모평균에서 멀리 떨어질 가능성이 **높아집니다**.
* **표본 평균들의 오차도 커집니다:** 모집단이 원래 변동성이 크면, 표본 평균들의 분포도 넓게 퍼지게 되므로, **표준 오차도 커집니다**.

**결론:** 모집단 표준편차($\sigma$)가 커지면 **분자**가 커지므로, 표준 오차는 **커집니다**.

---

###  정리

표준 오차 공식은 **표본 평균의 정확도**를 나타내는 아주 중요한 공식이며, 다음과 같은 의미를 담고 있습니다.

> **"오차는 모집단이 원래 가진 변동성에 비례하고 ($\sigma$), 표본을 많이 모을수록 ($\sqrt{n}$) 감소한다."**

---


## **표준 오차 ($\text{SE}$)로 나누는 이유**는 우리가 개별 데이터 포인트가 아닌 표본의 평균($\bar{x}$)을 검정하고 있기 때문.

##  Z-점수 공식의 핵심 차이


### 1. 개별 데이터의 Z-점수 (데이터 표준화)

이 공식은 개별 데이터 $x$가 **모평균 $\mu$**로부터 **모집단의 표준편차 $\sigma$**를 기준으로 얼마나 떨어져 있는지를 측정합니다.

$$\text{개별 Z-점수} = \frac{x - \mu}{\sigma}$$

* **사용 시점:** 특정 **개인**의 키가 전체 모집단 평균에서 얼마나 특이한지 알고 싶을 때.
* **나누는 값 ($\sigma$):** 모집단 데이터 하나하나의 퍼짐 정도.

---

### 2. 표본 평균의 Z-점수 (가설 검정)

이 공식은 우리가 추출한 **표본 평균 $\bar{x}$**가 **귀무가설의 모평균 $\mu_0$**으로부터 **표준 오차 ($\text{SE}$)**를 기준으로 얼마나 떨어져 있는지를 측정합니다.

$$\text{표본 평균 Z-점수} = \frac{\bar{x} - \mu_0}{\text{SE}} = \frac{\bar{x} - \mu_0}{\sigma / \sqrt{n}}$$

* **사용 시점:** 표본을 추출하여 구한 **평균($\bar{x}$)**이 모집단 평균($\mu_0$)과 **유의미하게 다른지** 검정할 때 (즉, 통계적 가설 검정).
* **나누는 값 ($\text{SE}$):** **표본 평균들의 분포**가 얼마나 퍼져 있는지를 나타내는 값.

---

##  표준 오차로 나누는 이유

표본 평균의 Z-검정에서 표준 오차($\text{SE} = \sigma / \sqrt{n}$)로 나누는 이유.

1.  **검정 대상이 개별 데이터가 아니기 때문입니다.**
    우리의 관심사는 개별 과자 무게가 100g과 얼마나 다른지가 아니라, **'표본 50개의 평균 (101.5g)'이 100g과 얼마나 다른가**이다.

2.  **표본 평균은 개별 데이터보다 변동성이 작기 때문.**
    표본의 평균은 개별 데이터보다 모평균 주변에 **더 촘촘하게** 모이는 경향이 있습다. 따라서, 표본 평균들의 퍼짐 정도는 개별 데이터의 퍼짐 정도($\sigma$)보다 **항상 작다.**

3.  **나누는 값은 검정 대상의 표준편차여야 합니다.**
    검정 대상인 **'표본 평균들의 분포'**의 표준편차는 $\sigma$가 아니라 **표준 오차 ($\text{SE}$)**. 따라서 표본 평균의 Z-점수를 계산할 때는 그에 해당하는 표준 오차로 나누어 주어야 정확한 통계적 해석이 가능.

쉽게 말해, **개별 데이터를 표준화**할 때는 개별 데이터의 표준편차인 $\sigma$로 나누고, **표본 평균을 표준화**할 때는 표본 평균의 표준편차인 $\text{SE}$로 나눈다고 이해하면됨.